lib/hamster/hash.rb

```diff
@@ -590,6 +590,27 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `hil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return value
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      else
+        if value.respond_to?(:dig)
+          value.dig(*rest)
+        end
+      end
+    end
```

Comment:
Great, makes sense to me! This also mean we could use this implementation as a `Dig` module included in hash and vector and get 2 for the price of one. Does that sound like a reasonable solution to you?